bot/utils/blurple_formatter.py

```diff
@@ -0,0 +1,717 @@
+                f"async{space(1)}def{space(1)}{name}{space()}({space()}{unparse(args, True)}{space()})"
+                + (f"{space()}->{space()}{unparse(returns)}" if returns else "")
+                + f"{space()}:",
+                *indent(body, randrange(2, 5)),
+            )
+        )
+    if isinstance(node, ast.AsyncWith):
+        items, body = node.items, node.body
+
+        return "\n".join(
+            [
+                f"async{space(1)}with {f',{space()}'.join(map(unparse, items))}{space()}:",
+                *indent(body, randrange(2, 5)),
+            ]
+        )
+    if isinstance(node, ast.Attribute):
+        owner, attr = node.value, node.attr
+
+        return f"{unparse(owner)}{space()}.{space()}{attr}"
+    if isinstance(node, ast.AugAssign):
+        target, op, value = node.target, node.op, node.value
+
+        return (
+            f"{unparse(target)}{space()}{op_strs[type(op)]}={space()}{unparse(value)};"
+        )
+    if isinstance(node, ast.Await):
+        value = node.value
+
+        return f"await{space(1)}{unparse(value)}"
+    if isinstance(node, ast.BoolOp):
+        op, values = node.op, node.values
+
+        return f"{space(1)}{op_strs[type(op)]}{space(1)}".join(
+            f"{(parenthesize if precedences[type(value)] <= precedences[type(op)] else unparse)(value)}"
+            for value in values
+        )
+    if isinstance(node, ast.Break):
+
+        return f"break{space()};"
+    if isinstance(node, ast.BinOp):
+        left, op, right = node.left, node.op, node.right
+
+        left_type = type(left)
+        right_type = type(right)
+        op_type = type(op)
+        if isinstance(left, (ast.BoolOp, ast.BinOp)):
+            left_type = type(left.op)
+        if isinstance(right, (ast.BoolOp, ast.BinOp)):
+            right_type = type(right.op)
+
+        par_left = (
+            precedences[left_type] < precedences[op_type]
+            if op_type not in r_assoc
+            else precedences[left_type] <= precedences[op_type]
+        )
+        par_right = (
+            precedences[right_type] <= precedences[op_type]
+            if op_type not in r_assoc
+            else precedences[right_type] < precedences[op_type]
+        )
+        nl_before = (op_type in nl_before_op) * "\n"
+        nl_after = (op_type not in nl_before_op) * "\n"
+        if nl_able:
+            return (
+                f"{(parenthesize if par_left else unparse)(left, nl_able)}{space()}"
+                + f"{nl_before}{op_strs[type(op)]}{space()}{nl_after}"
+                + f"{(parenthesize if par_right else unparse)(right, nl_able)}"
+            )
+        else:
+            return (
+                f"({(parenthesize if par_left else unparse)(left, True)}{space()}"
+                + f"{nl_before}{op_strs[type(op)]}{space()}{nl_after}"
+                + f"{(parenthesize if par_right else unparse)(right, True)})"
+            )
```

Comment:
What's the difference between these two? Just set `nl_able` to True and remove the latter codepath.

Reply:
resolved in 8c224a52be5000af380465a9d71a55997750fccd